### ISSUE TITLE
Weapon Selector: Index out of range

### ISSUE THREAD

**Issue by ChrisNZL** [ChrisNZL/Tallowmere2#438]
Reported in 0.0.18.

Feedback ID: 20200117-N96PW

**Comment by ChrisNZL** [comment]
Reported again – Feedback ID 20200203-ZG27P.

Probably something to do with selling an item to the Merchant, but the Weapon Selector is not getting updated.

```
7:55:52, Frame 19664, LOG »  Player Player ??? (Master1) is selling item to merchant... WeaponStruct Arcane Bow Master1 146
7:55:55, Frame 19806, LOG »  Player Player ??? (Master1) is selling item to merchant... WeaponStruct Lightning Orb Master683
7:55:57, Frame 19947, LOG »  Player Player ??? (Master1) is selling item to merchant... WeaponStruct Poison Boomerang Master1 052
7:55:58, Frame 20027, LOG »  Player Player ??? (Master1) is selling item to merchant... WeaponStruct Katana Master69
7:56:01, Frame 20189, LOG »  Player Player ??? (Master1) is selling item to merchant... ShieldStruct Serpent Shield Master65
7:56:03, Frame 20279, LOG »  Player Player ??? (Master1) is selling item to merchant... ShieldStruct Faded Shield Master50
7:56:09, Frame 20648, LOG »  Player Player ??? (Master1) is selling item to merchant... WeaponStruct Lightning Orb Master189

7:56:22, Frame 21444, EXCEPTION »  ArgumentOutOfRangeException: Index was out of range. Must be non-negative and less than the size of the collection.
Parameter name: index
>>>>> CRITICAL ERROR >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

System.ThrowHelper.ThrowArgumentOutOfRangeException (System.ExceptionArgument argument, System.ExceptionResource resource)
System.ThrowHelper.ThrowArgumentOutOfRangeException ()
T2.WeaponWheel2.SelectSlot (T2.WeaponWheel2Slot newSlot)
T2.WeaponWheel2.ProcessWeaponWheelInput (T2.Player player)
T2.Player.Update ()

GameStates: RompingThroughDungeon
GameSetupMode: Singleplayer
Players: 1
Room: 19 / 19
RoomModifiers: BoomerangsOnly (Boomerang Party), Tilted (Tilted), ExtraChests (Extra Chests)
```

**Comment by ChrisNZL** [comment]
Fixed in 0.1.0.

Was a cache/timing issue with item netIDs getting retrieved and cleared in a bad order.
